spec/inputs/http_poller_spec.rb

```diff
@@ -23,7 +23,11 @@
   }
   let(:klass) { LogStash::Inputs::HTTP_Poller }
 
-  subject { klass.new(default_opts) }
+  subject {
+    s = klass.new(default_opts)
+    s.register # We can't use a before block here because of rspec scoping
```

Comment:
The problem is that some of the tests use a different subject, so the before will run with a different subject. I could hoist that subject up but then there's more code duplication. Alternatively, I could make a helper function, but this seemed most clear.